org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
I took a look at this.

With this change in addCoreType

    _pgNameToJavaClass.put(pgArrayTypeName, "java.sql.Array");
    _pgNameToSQLType.put(pgArrayTypeName, Types.ARRAY);
    _pgNameToOid.put(pgArrayTypeName, arrayOid);
    pgArrayTypeName = "_" + pgTypeName;
    if (!_pgNameToJavaClass.containsKey(pgArrayTypeName)) {
      _pgNameToJavaClass.put(pgArrayTypeName, "java.sql.Array");
      _pgNameToSQLType.put(pgArrayTypeName, Types.ARRAY);
      _oidToPgName.put(arrayOid, pgArrayTypeName);
      _pgNameToOid.put(pgArrayTypeName, arrayOid);
    }

test behaves as expected.
It works because arrayoid is mapped to internal name. We can use same logic for custom type lookups.

Still, I personally would prefer if we registered arrayed for type[] version since that is more consistent.